src/lib/application.ts

```diff
@@ -312,25 +312,38 @@ export class Application extends ChildableComponent<Application, AbstractCompone
     }
 
     private generateConstrainedValues(str: any) {
-        let constrainedValues = [];
-
-        if (str && str['type'] && str['type'].type == 'union') {
-            if (str.type.types[1] && str.type.types[1].elementType && str.type.types[1].elementType.types) {
-                constrainedValues = str.type.types[1].elementType.types.map(function (type) {
-                    return type.value;
-                });
-                if (str.type.types[0].type && str.type.types[0].type.toLowerCase() == 'array') {
-                    var copy = [];
-                    for (var i = 0; i < constrainedValues.length; i++) {
-                        copy[i] = constrainedValues.slice(0, i + 1).join(',');
-                    }
-                    constrainedValues = copy;
-                }
-                constrainedValues = constrainedValues.slice(0, 4);
+        if (!str || !str['type'] || !(str['type'].type == 'union')) {
+            return [];
+        }
+
+        if (str.type.types[1] && str.type.types[1].elementType && str.type.types[1].elementType.types) {
+           return this.getConstrainedValuesFromElementType(str);
+        }
+
+        if (str.type.types) {
+            return this.getConstrainedValuesFromTypes(str);
+        }
+
+        return [];
+    }
+
+    private getConstrainedValuesFromElementType(str: any) {
+        let values = str.type.types[1].elementType.types.map(type => type.value);
+
+        if (str.type.types[0].type && str.type.types[0].type.toLowerCase() == 'array') {
+            var copy = [];
+            for (var i = 0; i < values.length; i++) {
+                copy[i] = values.slice(0, i + 1).join(',');
             }
+            values = copy;
         }
 
-        return constrainedValues;
+        return values.slice(0, 4);
```

Comment:
This block is identical in logic to what it was before (in the red diff). One question I had is why the decision to keep only the first 4 results?

The logic atm is to do the following:
Given union types `A | B | C` on an array option, the constrained values will be:
```
constrainedValues = [
    'A',
    'A,B',
    'A,B,C'
]
```
A second question maybe is why the gradual build up of values.

Note that in the IEditor is taking the last element of the array, splitting on the `,` and suggesting the entire list as autocompletions. But since we slice and return only the first 4 elements of the array, possibilities after #4 are lost atm.

Reply:
I truly cannot remember :D As explained, this was done in a hackathon a while ago, so I'm not 100% sure there's actually a good reason.

Reply:
Ok, I will remove the slice in order to return the whole array of values, allowing us to suggest the full list.